internal/app/admin/http/handler.go

```diff
@@ -0,0 +1,136 @@
+	for _, handler := range getHandlers(bootstrapConfig, orchestrator) {
+		http.Handle(handler.prefix, handler.handler)
+	}
+}
+
+func defaultHandler(handlers []Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		// The "/" pattern matches everything, so we need to check
+		// that we're at the root here.
+		if req.URL.Path != "/" {
+			http.NotFound(w, req)
+			return
+		}
+		fmt.Fprintf(w, "admin commands are:\n")
+		for _, handler := range handlers {
+			fmt.Fprintf(w, "  %s: %s\n", handler.prefix, handler.description)
+		}
+	}
+}
+
+func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		configString, err := interfaceToString(bootstrapConfig)
+		if err != nil {
+			fmt.Fprintf(w, "Failed to dump config: %s\n", err.Error())
+		}
+		fmt.Fprintf(w, "%s\n", configString)
+	}
+}
+
+// TODO(lisalu): Support dump of entire cache when no key is provided.
+// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
+func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		cacheKey := getCacheKeyParam(req.URL.Path)
+		cache := orchestrator.Orchestrator.GetReadOnlyCache(*o)
+		resource, err := cache.Fetch(cacheKey)
+		if err != nil {
+			fmt.Fprintf(w, "no resource for key %s found in cache.\n", cacheKey)
+			return
+		}
+		resourceString, err := resourceToString(resource)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "unable to convert resource to string.\n")
+			return
+		}
+		fmt.Fprint(w, resourceString)
+	}
+}
+
+// Converts an interface to a human-readable string for pretty-printing via marshalling to JSON.
+func interfaceToString(v interface{}) (string, error) {
+	prettyJSON, err := json.MarshalIndent(v, "", "  ")
+	if err != nil {
+		return "", err
+	}
+	return string(prettyJSON), nil
+}
+
+type marshallableResource struct {
+	Resp           *cache.Response
+	Requests       []*v2.DiscoveryRequest
+	ExpirationTime time.Time
+}
+
+// In order to marshal a Resource from the cache to JSON to be printed,
+// the map of requests is converted to a slice of just the keys,
+// since the bool value is meaningless.
+func resourceToString(resource *cache.Resource) (string, error) {
+	var requests []*v2.DiscoveryRequest
+	for request := range resource.Requests {
+		requests = append(requests, request)
+	}
+
+	resourceString := &marshallableResource{
+		Resp:           resource.Resp,
+		Requests:       requests,
+		ExpirationTime: resource.ExpirationTime,
+	}
+
+	return interfaceToString(resourceString)
+}
+
+func getCacheKeyParam(path string) string {
+	// Assumes that the URL is of the format `address/cache/parameter` and returns `parameter`.
+	splitPath := strings.SplitN(path, "/", 3)
+	return splitPath[2]
```

Comment:
Do we need to worry about accessing an out of bounds element here?

Reply:
This shouldn't happen based on how the server does route matching but I'll add a check in case this function is invoked elsewhere or the route matching changes in the future.